### PR TITLE
fix: support for excluding branches

### PR DESCRIPTION
**Pull request by salmanm**
Fixes https://github.com/nearform/github-action-check-linked-issues/issues/11

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,3 +1,6 @@
 import { run } from "./action.js";
+import * as util from "./util.js";
 
-run();
+if (util.shouldRun()) {
+  run();
```
Comment (simoneb) [inline src/index.js:5]: let's print a message explaining why we're not running for helping with debugging. let's maybe use annotations? no idea what they do but sounds appropriate https://github.com/actions/toolkit/tree/main/packages/core#annotations

Comment (salmanm) [reply src/index.js]: Done

Comment (simoneb) [reply src/index.js]: can't see it

Comment (salmanm) [reply src/index.js]: https://github.com/nearform/github-action-check-linked-issues/pull/12/files#diff-bfe9874d239014961b1ae4e89875a6155667db834a410aaaa2ebe3cf89820556R7

File: dist/index.js [hunk 2]
```diff
@@ -9727,9 +9734,14 @@ function shouldRun() {
 
 
 
-if (shouldRun()) {
-  run();
+
+
+if (!shouldRun()) {
+  core.notice("source branch matched the exclude pattern, exiting...");
```
Comment (simoneb) [inline dist/index.js:9740]: nit: to be precise in this function you don't know what the reason for not running the action is, but okay not a big deal

**Review by simoneb — APPROVED** [review]

File: dist/index.js (hunk 2 above)
Comment (salmanm) [reply dist/index.js:9740]: Yea I thought so too. tbh I wanted to convert this whole action to use `actions/github-script` and organise a bit better. But perhaps later.

**Pull request merged**